src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/Diagnostics/SubscriptionClientDiagnosticsTests.cs

```diff
@@ -162,5 +125,13 @@ protected void AssertRemoveRuleStop(string entityName, string eventName, object
                 Assert.Equal(removeRuleActivity, activity);
             }
         }
+
+        public void Dispose()
+        {
+            while (Activity.Current != null)
```

Comment:
Why did you have to add this? Who's leaking activities?

Reply:
That should not be there.   It was an intermittent step in removing the base class `Dispose` logic which I was able to eliminate by isolating the listener scope.  Thanks for the catch, this is going away!